pkg/controller/deployment/deployment_controller.go

```diff
@@ -457,6 +459,11 @@ func (dc *DeploymentController) syncRecreateDeployment(deployment extensions.Dep
 		return dc.updateDeploymentStatus(allRCs, newRC, deployment)
 	}
 
+	if deployment.Spec.RevisionHistoryLimit != nil {
+		// Cleanup old RCs
+		dc.cleanupOldRcs(oldRCs, deployment)
```

Comment:
Since you already log failed deletions I guess it's fine not to check for the error here